src/MicrosoftTeams.ts

```diff
@@ -421,22 +422,61 @@ namespace microsoftTeams
             let link = document.createElement("a");
             link.href = authenticateParameters.url;
 
-            let messageId = sendMessage(
-                "authentication.authenticate",
-                link.href,
-                authenticateParameters.width,
-                authenticateParameters.height);
-            callbacks[messageId] = (success: boolean, response: string) =>
+            // Clear any previous success/failure values
+            localStorage.removeItem("authentication.success");
+            localStorage.removeItem("authentication.failure");
+
+            // Open a popup window to the specified authentication URL
+            let authWindow: Window;
+            if (loadedInDesktopClient)
             {
-                if (success)
+                let messageId = sendMessage("authentication.authenticate", link.href);
+                callbacks[messageId] = (allowed: boolean) =>
                 {
-                    authenticateParameters.successCallback(response);
-                }
-                else
+                    if (allowed)
+                    {
+                        authWindow = currentWindow.open(link.href, "Microsoft Teams", "width = " + authenticateParameters.width + ", height = " + authenticateParameters.height);
```

Comment:
I will clean this up so that it follows the same logic as what we have in the FE client code today.